apina-core/src/test/kotlin/fi/evident/apina/spring/testclasses/ClassWithValueClasses.kt

```diff
@@ -0,0 +1,17 @@
+package fi.evident.apina.spring.testclasses
+
+@JvmInline
+value class ValueString(private val string: String)
+
+@JvmInline
+value class ValueInteger(private val integer: Int)
+
```

Comment:
I'm quite sure that the current code passes for this as well, but I think it's a good idea to add a test case for nested classes as well because its easy to imagine an implementation strategy where it would be easy to trip on this:

```kotlin
@JvmInline
value class ValueNested(private val v: ValueInteger)
```

So even though there's a level of indirection, the representation type should still be `Int`.

Reply:
I'll add this test case, it seems to work as expected.